pkg/compact/compact.go

```diff
@@ -901,7 +913,7 @@ func (cg *Group) areBlocksOverlapping(include *metadata.Meta, exclude ...*metada
 }
 
 // RepairIssue347 repairs the https://github.com/prometheus/tsdb/issues/347 issue when having issue347Error.
-func RepairIssue347(ctx context.Context, logger log.Logger, bkt objstore.Bucket, blocksMarkedForDeletion prometheus.Counter, issue347Err error) error {
+func RepairIssue347(ctx context.Context, logger log.Logger, bkt objstore.Bucket, blocksMarkedForDeletion prometheus.Counter, g *Group, issue347Err error) error {
```

Comment:
I prefer to pass `g.blockFilesConcurrency` or `UploadDirOption` in the function signature rather than passing the whole `Group` struct, WDYT?

Reply:
I think make sense...

I was just thinking "we may need more thing from the group in the future and if so, we will not need to change the signature".. but passing the `blockFilesConcurrency ` is fine as well.